design/automate_release_design.md

```diff
@@ -0,0 +1,204 @@
+
+`cyberark/secrets-provider-for-k8s` Jenkins pipeline
+
+1. Package the Secrets Provider Helm chart as part of Jenkins pipeline
+2. Dispatch an `version-release` event by sending a curl request with Helm package as part of request
+
+`cyberark/helm-charts` repository
+
+1. Listen on `version-release` events
+
+2. Once received, begin Github workflow by:
+
+   a. Creating and checking out a new branch
+
+   b. Commiting the Helm package to the branch
+
+   c. Creating a new PR and assigning reviewers
+
+### Additions to `cyberark/secrets-provider-for-k8s` Jenkins pipeline
+
+To handle the packaging the Helm Chart, we will [add a "Package artifacts" stage](https://github.com/cyberark/secrets-provider-for-k8s/pull/234/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R109) to the Jenkins pipeline that will run on *all* builds. This stage will be responsible for packaging the Helm Chart and saving the artifact to the `Artifacts` tabs of the build. An additional step will run on version builds (v1.1.0 for example) that will push those artifacts to `cyberark/helm-charts` repository. 
+
+The change resembles the following:
+
+```yaml
+stage('Package artifacts') {
+   steps {
+      sh 'ci/jenkins_build'
+        
+      archiveArtifacts artifacts: "helm-artifacts/", fingerprint: false, allowEmptyArchive: true
+	 }
+   when { tag "v*" }
+   steps {
+     sh 'ci/dispatch_version_release_event'
+   } 
+}
+```
+
+The  `ci/jenkins_build` script spins up a container and packages the Helm Chart like so:
+
+```bash
+source bin/build_utils
+helm_version=3.3.0
+
+docker run --rm \
+  -v $PWD/helm/secrets-provider:/root/helm/secrets-provider \
+  -v $PWD/helm-artifacts/:/root/helm-artifacts \
+  --workdir /root/helm-artifacts \
+   alpine/helm:${helm_version} package ../helm/secrets-provider
+```
+
+ The `ci/dispatch_version_release_event` script will only run on release builds. A curl request will be sent passing in the packaged Helm chart, dispatching an `version-release` event. This will trigger a workflow in `cyberark/helm-charts` to run.
+
+```bash
+curl -X POST \
+	-H "Authorization: token OAUTHN-TOKEN" \
+	"Accept: application/vnd.github.everest-preview+json" \
+	"Content-Type: application/javascript" \
+	https://api.github.com/repos/cyberark/helm-charts/dispatches
+	-d- '@/helm-artifacts/secrets-provider-v1.1.0.tgz'
+	-d '{"event_type": "version-release"}'
+```
+
+### Additions to `cyberark/helm-chart` Github repository
+
+In the `cyberark/helm-chart` repository we will need to create a Github workflow that listens for `version-release` events. Once received this will trigger a series of actions in response. The following is an incomplete Github workflow that will be added to the `cyberark/helm-chart` repository:
+
+```pseudocode
+name: Push Helm package workflow
+
+on:
+  repository_dispatch:
+    types: [version-release]
```

Comment:
we will need this in other projects too (eg for `homebrew-tools`). if `version-release` events are the kind of event that the `helm-charts` repo is listening for, should these instead be  called `helm-version-release` or `helm-chart-release`, so that the `helm-charts` project _knows_ its an event that it should be listening for?

to make this clearer, if I release secretless and want to dispatch a `version-release` event to update `homebrew-tools`, is the `helm-charts` repo going to also catch that event and try to do something with it, even though there is no helm chart update needed?

Reply:
I agree on the first part that the event should be helm-specific.

For your second part are you concerned that if we don't specify that there was a helm chart release event then another repo will catch an incorrect event?